### PR TITLE
fix: lifetimes

### PR DESCRIPTION
**Pull request by jayy-lmao**
- fix: format
- fix: lifetimes

### CODE REVIEW

File: src/simple_types.rs [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+impl<'a> OrderLineId<'a> {
+    pub(crate) fn new(id: &'a str) -> Self {
+        Self { value: id }
     }
 }
-#[derive(Clone)]
-pub(crate) struct OrderId {
-    value: String,
+#[derive(Clone, Copy)]
+pub(crate) struct OrderId<'a> {
```
Comment (Swoorup) [inline src/simple_types.rs:14]: I started using something like https://github.com/zbraniecki/tinystr 
and ensure it is struct allocated, so copy is just as fast :) 

Lifetime explosion can be a pain to deal with. 

Long time no see 😛

Comment (Swoorup) [reply src/simple_types.rs:14]: There is also https://lib.rs/crates/shared-string perhaps more suited for larger strings.

Comment (jayy-lmao) [reply src/simple_types.rs:14]: Thank you for the help! These crates seem really useful, might see if they're fit for my work stuff too. 

What's brought you to this repo?

Comment (Swoorup) [reply src/simple_types.rs:14]: Was checking out Rust to get a feel for how productive I can be. I ended up doing a port of my pet Scala project to Rust for perf reasons 😄 

Was looking for ideas and design patterns. 😄

Comment (Swoorup) [reply src/simple_types.rs:14]: You could use macros for some of the type constraints too. I have created a macro which auto generates, `Mul`, `Div`, `From`, `To` impls

```rust
num_new_type_rules![
  // commutative
  UnitPrice * Quantity => Amount
  Quantity * Amount => Amount

  // division
  Amount / UnitPrice => Quantity
  Amount / Quantity => UnitPrice

  // from-into conversion
  Amount <=> Quantity
  Amount <=> UnitPrice
];
```

**Pull request closed without merging**